### PR TITLE
POD-717 | Support signing commits with SSH keys 

### PR DESCRIPTION
**Pull request by janekbaraniewski**
This PR adds support for signing git commits inside DevPod using ssh keys.
We already support signing commits with gpg keys. This adds missing part of ssh keys support.

Issue: #1146 

# Usage

## Default way

By default DevPod will detect if user has ssh signatures set up in local git. This is done based on `git.gpg.format` git setting on users machine. If this is set to `ssh` DevPod will setup ssh signature support.

## Manual way

There is a way to manually enable ssh signatures when running `devpod up` :
```
devpod up github.com/loft-sh/devpod --git-ssh-signing-key /Users/johndoe/.ssh/mykey.pub
```

### CODE REVIEW

File: pkg/agent/tunnelserver/tunnelserver.go [hunk 1]
```diff
@@ -216,6 +217,26 @@ func (t *tunnelServer) GitCredentials(ctx context.Context, message *tunnel.Messa
 	return &tunnel.Message{Message: string(out)}, nil
 }
 
+func (t *tunnelServer) GitSSHSignature(ctx context.Context, message *tunnel.Message) (*tunnel.Message, error) {
```
Comment (pascalbreuninger) [inline pkg/agent/tunnelserver/tunnelserver.go:220]: You'll also need to implement this function in ./proxyserver (just forward the message) for Pro setups

File: cmd/ssh.go [hunk 2]
```diff
@@ -472,6 +473,12 @@ func (cmd *SSHCmd) startProxyServices(
 		if gitCredentials {
 			command += " --configure-git-helper"
 		}
+
+		// check if we should enable git ssh commit signature support
+		if cmd.GitSSHSignatureForwarding || devPodConfig.ContextOption(config.ContextOptionGitSSHSignatureForwarding) == "true" {
```
Comment (pascalbreuninger) [inline cmd/ssh.go:478]: What happens if gpg agent forwarding is enabled as well?

Comment (pascalbreuninger) [reply cmd/ssh.go:478]: And by setting this we enable it for everyone even though they might not have set it up locally 🤔
Imo the intuitive way would be to enable it if it's available locally but opt-in if not. What do you think?

Comment (janekbaraniewski) [reply cmd/ssh.go:478]: yeah that's the idea, working on it atm. For testing so far it was just easier for me to have it enabled all the time. That's how I wanted to set it up:
- if you have ssh signing set up locally we'll automatically set it up for you 
- if you don't you'll be able to specify a flag with path to key you want to use for signing and we'll set it up 

And about the gpg agent forwarding - it wouldn't matter, as long as `gpg.format=ssh` is set git will use our tool. So gpg agent would be just waiting there but wouldn't be used. I'll do some testing though

**Pull request merged**